src/commands/compress.rs

```diff
@@ -55,6 +55,10 @@ pub fn compress_files(
                 encoder,
                 level.map_or_else(Default::default, |l| bzip2::Compression::new((l as u32).clamp(1, 9))),
             )),
+            Bzip3 => Box::new(
+                // Unwrap is safe when using a valid block size
+                bzip3::write::Bz3Encoder::new(encoder, bytesize::ByteSize::mib(5).0 as usize).unwrap(),
+            ),
```

Comment:
This needs to throw an error with `?` instead of using `unwrap()`.

Compiler is going to immediately complain, so you add this code: https://github.com/ouch-org/ouch/compare/main...bzip3#diff-97e25e2a0e41c578875856e97b659be2719a65227c104b992e3144efa000c35eR175-R187

And it will handle the error conversion.

The `BlockSize` should be unreachable if you're providing `mib(5)`, but just to be safe, please remove the `unreachable!` and add the `BlockSize` in the other treatment arm:

```diff
- Bz3Error::BlockSize => unreachable!(),
- Bz3Error::ProcessBlock(_) | Bz3Error::InvalidSignature => {
+ Bz3Error::BlockSize | Bz3Error::ProcessBlock(_) | Bz3Error::InvalidSignature => {
      FinalError::with_title("bzip3 error").detail(err.to_string()).into()
  }

Reply:
Thank you for the feedback. I was not aware of your branch, looks like I could have saved some time ;) Anyway, I implemented your changes in [2289def](https://github.com/ouch-org/ouch/pull/522/commits/2289def77b29b6da11e277763826c82bbd41f7d3). Since they merged your PR over at [bzip3-rs](https://github.com/bczhc/bzip3-rs) I used this commit instead, at least until they publish a new crate version.

Reply:
> looks like I could have saved some time ;)

Yep xD, I hope you enjoyed doing the coding tho.

I'll keep in mind to always create an issue or a draft PR to signal this to other people.